tests/Microsoft.Identity.Web.Test.Common/Mocks/MockHttpContextAccessor.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Identity.Web.Test.Common.Mocks
+{
+    public class MockHttpContextAccessor : IHttpContextAccessor
```

Comment:
Why not use NSubstitute to create mocks on the fly? You will have better control over them.